rct229/rulesets/ashrae9012019/section19/section19rule24.py

```diff
@@ -0,0 +1,87 @@
+from rct229.rule_engine.rule_base import RuleDefinitionBase
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.rulesets.ashrae9012019.data.schema_enums import schema_enums
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_proposed_hvac_modeled_with_virtual_cooling import (
+    get_proposed_hvac_modeled_with_virtual_cooling,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_proposed_hvac_modeled_with_virtual_heating import (
+    get_proposed_hvac_modeled_with_virtual_heating,
+)
+from rct229.utils.pint_utils import ZERO
+
+FAN_SYSTEM_OPERATION = schema_enums["FanSystemOperationOptions"]
+
+
+class Section19Rule24(RuleDefinitionListIndexedBase):
+    """Rule 24 of ASHRAE 90.1-2019 Appendix G Section 19 (HVAC - General)"""
+
+    def __init__(self):
+        super(Section19Rule24, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, False, True),
+            each_rule=Section19Rule24.HVACRule(),
+            index_rmr="proposed",
+            id="19-24",
+            description="Schedules for HVAC fans that provide outdoor air for ventilation shall run continuously whenever spaces are occupied in the proposed design.",
+            ruleset_section_title="HVAC - General",
+            standard_section="Section G3.1-4 Schedules for the proposed building excluding exception #1 and Section G3.1.2.4.",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_instances/0",
+            list_path="$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+        )
+
+    def create_data(self, context, data):
+        rmi_p = context.proposed
+
+        inapplicable_hvac_sys_list_p = list(
+            set(
+                get_proposed_hvac_modeled_with_virtual_cooling(rmi_p)
+                + get_proposed_hvac_modeled_with_virtual_heating(rmi_p)
+            )
+        )
+
+        return {"inapplicable_hvac_sys_list_p": inapplicable_hvac_sys_list_p}
+
+    class HVACRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section19Rule24.HVACRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, False, True),
+                required_fields={
+                    "$": ["fan_system"],
+                    "fan_system": [
+                        "operation_during_occupied",
+                    ],
+                },
+            )
+
+        def is_applicable(self, context, data=None):
+            hvac_p = context.proposed
+            hvac_id_p = hvac_p["id"]
+            inapplicable_hvac_sys_list_p = data["inapplicable_hvac_sys_list_p"]
+
+            return hvac_id_p not in inapplicable_hvac_sys_list_p
+
+        def get_calc_vals(self, context, data=None):
+            hvac_p = context.proposed
+
+            operation_during_unoccupied_p = hvac_p["fan_system"][
```

Comment:
do we want operation_during_occupied instead?

Reply:
I'm a bit confused. Don't we get the ```operation_during_unoccupied``` key from the ```fan_system```? So, isn't ```operation_during_unoccupied_p``` correct?

Reply:
the rule is verifying continuous fan_system operation during occupied hours, not unoccupied hours. RDS also references the operation_during_occupied key

Reply:
Addressed.